test/exe/run_test.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+require 'test_helper'
+require 'krane/cli/krane'
+
+class RunTest < KubernetesDeploy::TestCase
+  def test_run_with_default_options
+    set_krane_run_expectations
+    krane_run!
+  end
+
+  def test_run_parses_global_timeout
+    set_krane_run_expectations(new_args: { max_watch_seconds: 10 })
+    krane_run!(flags: '--global-timeout 10s')
+    set_krane_run_expectations(new_args: { max_watch_seconds: 60**2 })
+    krane_run!(flags: '--global-timeout 1h')
+  end
+
+  def test_run_parses_verify_result
+    set_krane_run_expectations(run_args: { verify_result: true })
+    krane_run!(flags: '--verify-result true')
+    set_krane_run_expectations(run_args: { verify_result: false })
+    krane_run!(flags: '--no-verify-result')
+  end
+
+  def test_run_parses_command
+    set_krane_run_expectations(run_args: { entrypoint: %w(/bin/sh) })
+    krane_run!(flags: '--command /bin/sh')
+  end
+
+  def test_run_parses_arguments
+    set_krane_run_expectations(run_args: { args: %w(hello) })
+    krane_run!(flags: '--arguments hello')
+  end
+
+  def test_run_parses_template
+    set_krane_run_expectations(run_args: { task_template: 'some-name' })
+    krane_run!(flags: '--template some-name')
+  end
+
+  def test_run_parses_env_vars
+    set_krane_run_expectations(run_args: { env_vars: %w(SOMETHING=8000 FOO=bar) })
+    krane_run!(flags: '--env-vars SOMETHING=8000,FOO=bar')
+  end
+
+  def test_run_failure_as_black_box
```

Comment:
What is actually being tested here? (e.g. that the exit status is 1, `-q` isn't a valid arg, ...)

Reply:
That's it. Test that it fails and returns status 1 when it's used incorrectly (see [here](https://github.com/Shopify/kubernetes-deploy/blob/master/test/exe/restart_test.rb#L42-L47) for where I knocked this off from heh)

I could make the test less arbitrary though, for sure. I'll make it more realistic and add tests failing on my own flags 😛

Reply:
If you're just testing the right exit code I think this is good. I just wanted to make sure we are testing our code and not thor.

Reply:
Fair!